dissect/target/plugins/os/windows/dpapi/crypto.py

```diff
@@ -140,158 +114,135 @@ def __init_subclass__(cls):
     def from_id(cls, id: int) -> HashAlgorithm:
         return HASH_ALGORITHMS[id]()
 
+    @classmethod
     def from_name(cls, name: str) -> Optional[HashAlgorithm]:
-        res = [i for i in HASH_ALGORITHMS.values() if i.name == name]
-        return res[0] if res else None
-
-    @property
-    @abstractproperty
-    def id(self):
-        raise NotImplementedError
-
-    @property
-    @abstractproperty
-    def name(self):
-        raise NotImplementedError
+        return HASH_ALGORITHMS[name]()
```

Comment:
Did you mean to initialize `HASH_ALGORITHMS[name]` in `__init_subclass__`?

Reply:
Yes, fixed!